ecs-deploy

```diff
@@ -135,6 +137,61 @@ function deployService() {
   printf "Done deploying service %s to %s...\n" $SERVICE $CLUSTER
 }
 
+execCommand() {
+  printf "Executing command '%s' in service %s of %s...\n" "$COMMAND" $SERVICE $CLUSTER
+
+  EXEC_SUCCESS=false
+
+  current_task_arns="$(
+    $ECS_CLI list-tasks $DEBUG \
+      --cluster "$CLUSTER" \
+      --family "$SERVICE" \
+      --query 'taskArns' |
+      jq -r '.[]'
+  )"
+  container_instances="$(
+    $ECS_CLI describe-tasks $DEBUG \
+      --cluster "$CLUSTER" \
+      --tasks $current_task_arns \
+      --query 'tasks[*].containerInstanceArn' |
+      jq -r '.[]'
+  )"
+  for container in $container_instances; do
+    ec2_instance_id=$(
+      $ECS_CLI describe-container-instances $DEBUG \
+        --cluster "$CLUSTER" \
+        --container-instances "$container" \
+        --region eu-central-1 \
+        --query 'containerInstances[*].ec2InstanceId' |
+        jq -r '.[]'
+    )
+    docker_ps_output=$(
+      $SSM_CLI start-session $DEBUG \
+        --target $ec2_instance_id \
+        --document-name 'AWS-StartNonInteractiveCommand' \
+        --parameters '{"command": ["sudo docker ps --filter name='"$SERVICE"'"]}'
+    )
+    container_id=$(echo $docker_ps_output | cut -d ' ' -f 14)
+
+    printf "Starting session in docker container %s of EC2 instance %s...\n" "$COMMAND" $container_id $ec2_instance_id
+
+    $SSM_CLI start-session $DEBUG \
+      --target $ec2_instance_id \
+      --document-name 'AWS-StartNonInteractiveCommand' \
+      --parameters '{"command": ["sudo docker exec '"$container_id"' '"$COMMAND"'"]}'
+
+    EXEC_SUCCESS=true
```

Comment:
It seems the command is considered executed successfully if it just has been executed, regardless whether the exit code was `0` or not. Do you know if there is a way to capture the exit code? I didn't look closely into it, but there is this open issue https://github.com/aws/containers-roadmap/issues/2207. While it might not be critical for us now, it is good to keep this in mind and maybe document it in the README.md

Reply:
This should now work by using `ssm send-command` instead of `ssm start-session`, see https://github.com/formunauts/drone-ecs-deploy-advanced/pull/1/commits/1c6eaa457d7c14c9b025ae3b63af0f0a354afd54.